sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/ViewFn.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.transforms;
+
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PCollectionView;
+
+import java.io.Serializable;
+
+/**
+ * A function to adapt a primitive "view" of a {@link PCollection} - some materialization
+ * specified in the Beam model and implemented by the runner - to a user-facing view type
+ * for side input.
+ *
+ * <p>Both the underlying primitive view and the user-facing view are immutable.
+ *
+ * <p>The most common case is using the {@link View} transforms to prepare a {@link PCollection}
+ * for use as a side input to {@link ParDo}. See {@link View#asSingleton()},
+ * {@link View#asIterable()}, and {@link View#asMap()} for more detail on specific views
+ * available in the SDK.
+ *
+ * @param <PrimitiveViewT> the type of the underlying primitive view, provided by the runner
+ *        <ViewT> the type of the value(s) accessible via this {@link PCollectionView}
+ */
+public abstract class ViewFn<PrimitiveViewT, ViewT> implements Serializable {
```

Comment:
I wasn't sure where to fall with this. We have tended towards abstract classes to be able to add default implementations of new methods. But with the new runner/SDK relationship, perhaps strong documentation would be enough and we could go with the better architecture of an interface + default abstract parent class.